pkg/controller/service/controller.go

```diff
@@ -334,6 +334,13 @@ func (s *Controller) syncLoadBalancerIfNeeded(service *v1.Service, key string) (
 	} else {
 		// Create or update the load balancer if service wants one.
 		op = ensureLoadBalancer
+		if !servicehelper.HasLBFinalizer(service) && len(service.ObjectMeta.Finalizers) > 0 {
```

Comment:
This is pretty broad. We may want to think about what this means for users that use finalizers that we don't know about.

Reply:
I agree. I didn't want to put checks for cloud-provider specific finalizer keys here either.
Any users that want service controller to manage the services would keep the default service controller finalizer. This check will be true if the default finalizer is NOT present and some other finalizer key is.

Reply:
Seems strange that adding a random finalizer could disable the entire code path. Does this have to live in the service controller layer? We can check explicitly if it is within gce provider.